guides/release/routing/specifying-a-routes-model.md

```diff
@@ -163,11 +175,118 @@ each record in the song model and album model:
+
+### Creating a dynamic segment
+
+A dynamic segment can be created using Ember CLI:
+
+```bash
+ember generate route photo --path "photos/:id"
+```
+
+This command will create a `.js` file, an `.hbs` file, and add a route in `router.js`:
+
+```javascript {data-filename=app/router.js}
+Router.map(function() {
+  this.route('photo', { path: '/photos/:id' });
+});
+```
+
+Whatever shows up in the URL at the `:id`, the dynamic segment, will be available in the params for the route's `model` hook.
+
+```javascript {data-filename=app/routes/photo.js}
+import Route from '@ember/routing/route';
+
+export default Route.extend({
+  model(params) {
+    console.log('This came from the URL: ' + params.id)
+  }
+});
+```
+
+The `:id` name here isn't something special.
+Whatever name you give the dynamic segment, that's the key it will show up as in the `model` parameters.
+You could call it `:puppies` in `router.js` if you wanted to, and access it in the model as `params.puppies`.
+
+In the `model` hook for routes with dynamic segments, it's your job to
+use the ID (something like `47` or `post-slug`) when you make a data request.
+
+For example, if your app uses Ember Data to make API requests, you could include the ID in a `findRecord`:
+
+```javascript {data-filename=app/routes/photo.js}
+import Route from '@ember/routing/route';
+
+export default Route.extend({
+  model(params) {
+    this.store.findRecord('photo', params.id)
+  }
+});
+```
+
+### Linking to a dynamic segment
+
+There are two ways to link to a dynamic segment from an `.hbs` template using the
+[link-to](../../templates/links/)
+helper.
+Depending on which approach you use, it will affect whether that route's model hook is run.
+To learn how to link to a dynamic segment from within the JavaScript file, see the API documentation on
+[`transitionTo`](https://api.emberjs.com/ember/3.8/classes/RouterService/methods/transitionTo?anchor=transitionTo)
+instead.
+
+When you provide a string or number to the `link-to`, the dynamic segment's model hook will run when the app transitions to the new route.
+In this example, `photo.id` might have a value that's an ID like `4`:
+
+```handlebars
+{{#each model as |photo|}}
+  {{#link-to "photo" photo.id}}
+    link text to display
+  {{/link-to}}
+{{/each}}
+```
+
+However, if you provide the entire model context, the model hook for that URL segment will _not_ be run. In this example, we are passing the entire `photo` record:
+
+```handlebars
+{{#each model as |photo|}}
+  {{#link-to "photo" photo}}
+    link text to display
+  {{/link-to}}
+{{/each}}
+```
+
+If a route you are trying to link to has multiple dynamic segments, be sure to specify all the necessary information for each segment. For example, a route like `/photos/4/comments/18` would need:
+
+```handlebars
+{{#link-to "photos.photo.comments.comment" 4 18}}
+  link text to display
+{{/link-to}}
+```
+
+Routes without dynamic segments will always execute the model hook.
+
```

Comment:
An additional common thing that trolls people, which I didn't see is mention of, the default route behavior for dynamic segments like formatted like `:book_id`.

For a route with no model hook implemented (even a route with no `.js` file at all), the default implementation of `model()` will be equivalent to

```js
model(params) {
 return this.store.find('book', params.book_id);
}
```

Reply:
Wait, what? It automatically provides a model hook that retrieves data? 😱

Reply:
yea this is something that I've known about for a long time but it's never been documented (that I know of) so it is completely alien to me 🙈 I have never relied on it once since forever

Reply:
Added a note on this. Thanks for the info!